jdbc-sink/src/main/java/org/springframework/cloud/stream/module/jdbc/JdbcSinkConfiguration.java

```diff
@@ -0,0 +1,217 @@
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.DirectFieldAccessor;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.ConfigurationPropertiesBinding;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.EvaluationException;
+import org.springframework.expression.Expression;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.integration.context.IntegrationContextUtils;
+import org.springframework.integration.jdbc.JdbcMessageHandler;
+import org.springframework.integration.jdbc.SqlParameterSourceFactory;
+import org.springframework.integration.json.JsonPropertyAccessor;
+import org.springframework.jdbc.core.JdbcOperations;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.jdbc.datasource.init.DataSourceInitializer;
+import org.springframework.jdbc.datasource.init.ResourceDatabasePopulator;
+import org.springframework.messaging.Message;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+/**
+ * A module that writes its incoming payload to an RDBMS using JDBC.
+ *
+ * @author Eric Bottard
+ * @author Thomas Risberg
+ */
+@EnableBinding(Sink.class)
+@EnableConfigurationProperties(JdbcSinkProperties.class)
+public class JdbcSinkConfiguration {
+
+	private static final Logger logger = LoggerFactory.getLogger(JdbcSinkConfiguration.class);
+
+	public static final Object NOT_SET = new Object();
+
+	@Autowired(required = false)
+	private SpelExpressionParser spelExpressionParser = new SpelExpressionParser();
+
+	@Autowired
+	private BeanFactory beanFactory;
+
+	protected EvaluationContext evaluationContext;
+
+	@Autowired
+	private JdbcSinkProperties properties;
+
+	@Autowired
+	private JdbcMessageHandler msgHandler;
+
+
+	@Bean
+	public JdbcMessageHandler jdbcMessageHandler(DataSource dataSource) {
+		final MultiValueMap<String, Expression> columnExpressionVariations = new LinkedMultiValueMap<>();
+		for (Map.Entry<String, String> entry : properties.getColumns().entrySet()) {
+			String value = entry.getValue();
+			columnExpressionVariations.add(entry.getKey(), spelExpressionParser.parseExpression(value));
+			if (!value.startsWith("payload")) {
```

Comment:
I suggest for case of `payload` always request this prefix:
```
columns=a: payload.a, b
```
where `b` will be converted to the `b: payload.b`, like you do right now.
Only single `expression` and, therefore, `ExpressionEvaluatingSqlParameterSourceFactory` gain.
Just because I will be able to do this:
```
columns=a: payload.a, b: headers.b
```